lib/DynamoRequest.js

```diff
@@ -95,32 +80,52 @@ DynamoRequest.prototype.setReturnValues = function (returnValues) {
   return this
 }
 
-DynamoRequest.prototype.setFilter = function (filter) {
-  if (filter) {
-    this.data.ScanFilter = {}
-    for (var key in filter) {
-      var op = filter[key][0]
-      var attrList = null
-      if (op === 'BETWEEN') {
-        attrList = [typeUtil.valueToObject(filter[key][1]), typeUtil.valueToObject(filter[key][2])]
-      } else if (op === 'IN') {
-        attrList = []
-        for (var i = 1; i < filter[key].length; i += 1) {
-          attrList.push(typeUtil.valueToObject(filter[key][i]))
-        }
-      } else if (op === 'NULL' || op === 'NOT_NULL') {
-        attrList = null
-      } else {
-        // All other ops have 1 argument.
-        attrList = [typeUtil.valueToObject(filter[key][1])]
-      }
+/**
+ * @param {Array.<FilterBuilder>} conditions An array of conditions, possibly null to indicate
+ *     no conditions.
+ * @return {DynamoRequest}
+ */
+DynamoRequest.prototype.setKeyConditions = function (conditions) {
+  this.data.KeyConditions = {}
+  if (conditions) {
+    assert.ok(Array.isArray(conditions), 'Expected array')
+    for (var i = 0; i < conditions.length; i += 1) {
+      assert.ok(conditions[i] instanceof FilterBuilder, 'Expected FilterBuilder')
```

Comment:
`FilterBuilder` unfortunately.  `ConditionBuilder` builds a structure that can only be used with `Expects`.  Eventually, I'd like to change `Expects` to use the output of `FilterBuilder` (since it can) and remove `ConditionBuilder`, but that would change existing behavior so I'd like to save that for a future PR.